src/Microsoft.Data.SqlClient/src/Microsoft/Data/ProviderBase/DbConnectionPool.cs

```diff
@@ -1292,13 +1316,16 @@ private bool TryGetConnection(DbConnection owningObject, uint waitForMultipleObj
                 do
                 {
                     int waitResult = BOGUS_HANDLE;
-
+#if NETFRAMEWORK
                     RuntimeHelpers.PrepareConstrainedRegions();
+#endif
                     try
                     {
+#if NETFRAMEWORK
                         // We absolutely must have the value of waitResult set, 
                         // or we may leak the mutex in async abort cases.
                         RuntimeHelpers.PrepareConstrainedRegions();
+#endif
                         try
```

Comment:
What is the point of an empty try block?

Reply:
In .NET Core, it's a no-op. In .NET Framework, it's a Constrained Execution Region.

Reply:
Interesting 🤔 I'll have to give this a deeper read.